data_loader.py

```diff
@@ -28,45 +31,103 @@ def disconnect(self):
         # Nothing to do
         pass
 
+    def _flatten_dict(self, data: dict, keystring: str = ""):
+        """Sufficent flattening dict"""
+        if type(data) == dict and len(data) != 0:
+            keystring = keystring + "." if keystring else keystring
+            for k in data:
+                yield from self._flatten_dict(data[k], keystring + str(k))
+        else:
+            yield keystring, data
+
+    def _combine_excel(self, list_of_articles: list):
+        """Add excel values to all articles"""
+        review_status = pd.read_excel(
+            self.args.reference_data_file,
+            sheet_name="review_status",
+            header=2,
+            usecols="B:E",
+            dtype=str,
+        )
+        date_completed = pd.read_excel(
+            self.args.reference_data_file, sheet_name="date_completed"
+        )
+
+        # Preprocessing data
+        review_status.columns = review_status.columns.str.replace(" ", "_").str.lower()
+        date_completed.columns = date_completed.columns.str.replace(
+            " ", "_"
+        ).str.lower()
+        review_status["article_id"] = review_status["article_id"].apply(
+            lambda x: x.strip()
+        )
+        date_completed["reference_id"] = date_completed["reference_id"].values.astype(
+            str
```

Comment:
Seems like it is good to have it inside method called `_preprocess_data`

Reply:
I don't know, but putting `dtype=str` in the `pd.read_excel` method does not solve the problem with `astype`?